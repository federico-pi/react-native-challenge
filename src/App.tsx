import React, { useEffect } from 'react';

import RNBootSplash from 'react-native-bootsplash';
import { QueryClient, QueryClientProvider } from 'react-query';

import { Main } from '~core/Main';

// The React Query client for caching the data
const queryClient = new QueryClient();

/**
 * The root app component
 */
const App = () => {
  useEffect(() => {
    const init = async () => {
      // Performing multiple sync and async function before dismissing the splashscreen
    };

    init().finally(() => {
      let timeout: ReturnType<typeof setTimeout>;

      timeout = setTimeout(
        async () => await RNBootSplash.hide({ fade: true }),
        1750
      );

      return () => clearTimeout(timeout);
    });
  }, []);

  return (
    <QueryClientProvider client={queryClient}>
      <Main />
    </QueryClientProvider>
  );
};

export default App;
