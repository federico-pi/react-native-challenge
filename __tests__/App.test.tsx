import React from 'react';

import renderer, { create, act } from 'react-test-renderer';

import App from '../src/App';

it('renders correctly with defaults', async () => {
  const app = renderer.create(<App />).toJSON();

  expect(app).toMatchSnapshot();
});

it('has 1 child', async () => {
  let tree;
  act(() => {
    tree = create(<App />);
  });

  expect(tree?.toJSON().children.length).toBe(1);
});
