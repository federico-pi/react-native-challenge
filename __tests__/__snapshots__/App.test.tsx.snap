// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 320,
                  "justifyContent": "space-between",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 32,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                  "width": 702,
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder="home:placeholder"
                style={
                  Object {
                    "color": "#191D30",
                    "flex": 1,
                    "fontFamily": "Poppins-SemiBold",
                    "fontSize": 16,
                  }
                }
              />
              <View
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#848A94",
                    "borderRadius": 16,
                    "height": 32,
                    "justifyContent": "center",
                    "marginLeft": 12,
                    "opacity": 1,
                    "padding": 10,
                    "width": 32,
                    "zIndex": 2,
                  }
                }
              >
                <SvgMock
                  stroke="#fff"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
